docs/overview.rst

```diff
@@ -0,0 +1,97 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Overview
+========
+Invenio-Files-REST is a files storage module. It allows you to store and retrieve
+files in a similar way to Amazon S3 APIs.
+
+In order to better understand what you can achieve with this Invenio module,
+the following overview will introduce you to its key concepts and terminology.
+
+In Invenio-Files-REST, a file is represented by an abstraction called :code:`Object`.
+An Object acts like container for a particular file (as identified by its name),
+and holds *it* as well as all its previous versions (if any). The latest version
+of the file is referred to as the :code:`HEAD`, while a version of the file is
+referred to as an :code:`Object Version`. The link between an :code:`Object Version`
+and the actual file on disk is made by a :code:`File Instance`. What this allows
+is for multiple :code:`Object Versions` to point to the same :code:`File Instance`,
+allowing some operations to be performed more efficiently, such as snapshots
+without duplicating files or migrating data.
+Just as in a computer files are contained inside folders, each :code:`Object` has
+to be contained in a :code:`Bucket`. The bucket is identified by a unique ID,
+assigned automatically at creation. A :code:`Bucket` is created by default in the
+default :code:`Location`, however that can be changed such that when creating a
+:code:`Bucket`, a particular :code:`Location` for it can be specified. The
+:code:`Bucket` can also have a maximum quota assigned to it, and an important
+point to note is that the :code:`Objects` inside it do not necessarily have to
+be located in the same :code:`Location`. The :code:`Location` can be used to
```

Comment:
> and an important point to note is that the :code:`Objects` inside it do not necessarily have to
be located in the same :code:`Location`

Hey! Are you sure about that? The `Bucket` is the one that has the `Location` as a thing handled by Invenio-Files-Rest, I understand that you can set the "physical location" of a file using [`ObjectVersion.set_location`](https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/models.py#L1006) but I think this is used on the special case where the file is outside "our file system", i.e. linked.
I might be wrong ☺️